README.md

```diff
@@ -1,2 +1,65 @@
-# java-api-assessment
-Java API assessment template (intermediate level)
+# **API Development with Java & SpringBoot Assessment**
+
+## **Introduction**
+Dive into the world of API development using Java & SpringBoot. We're handing over a skeleton codebase; your challenge is to shape a top-notch API from it.
+
+You can build any API of your choosing, but it must include the following:
+
+1. At least one algorithm
+1. Unit test at least one class
+1. Store the data in a JSON file 
+1. Exception handling 
+1. Evidence of inheritance
+1. Good use of HTTP Protocols - methods, request and response, have full CRUD operations supported 
+1. Documentation
+
+### **Learning Outcomes:**
+
+By the end of this assessment, you should be able to:
+
+1. **Design and Architect APIs**: Get to grips with the nitty-gritty of curating a top-quality API, focusing on data flow and endpoint interactions.
+1. **Implement Best Practices**: Showcase your adherence to Java & SpringBoot coding standards, error handling, and optimal project structure.
+1. **Code Integration**: Seamlessly combine your creations with the provided skeleton codebase.
+1. **Exception Management**: Efficiently handle exceptions, ensuring your API remains sturdy and dependable.
+
+Onward with this assessment, you're set for a deep dive into API development with Java & SpringBoot.
+
+## **Design & Requirements**
+
+### **Design Considerations:**
+- **API Flow**: Map out your API's progression, from endpoints to their functionalities.
+
+### **Requirements List:**
+- **Core**: Make use of Java & SpringBoot.
+- **End Points**: Ensure they are detailed and fully operational.
+- **Error Handling**: Your API should handle mishaps gracefully and return informative feedback.
+
+### **Learning Outcomes:**
+- Acknowledge the pivotal role of a focused design in APIs.
+- See firsthand how a detailed requirements list can pave the way for successful development.
+
+## **Repository Management**
+
+- **Consistent Commits**: Commit often, capturing your progress and thought process.
+- **README**: Not just an afterthought. Fill it with the essence of your API, setup instructions, and other salient details.
+
+### **Learning Outcomes:**
+- Hone your skills in effective version control.
+- Recognise the value of a well-curated repository.
+
+## **Code Quality & Structure**
+
+- **Best Practices**: Stick to Java & SpringBoot best practices and conventions.
+- **Modularity**: Your code should be modular, reusable, and easily comprehensible.
+
+#### **Learning Outcomes:**
+- Craft clean, efficient, and maintainable code.
+- Harness Java & SpringBoot to its fullest.
+
+---
+
+## **Deliverables / What You Should Submit:**
+1. **Repository Link**: A GitHub link showing off all your handiwork.
+1. **Documentation**: A top-tier README.md in your repository, going over setup, endpoints, and any other crucial bits and bobs.
+
```

Comment:
Does this need a "Getting Started" section with initial steps on how to build and run the app?

Reply:
Good shout 👍🏾